include/s3select.h

```diff
@@ -1811,15 +1808,15 @@ class csv_object : public base_s3object
   bool m_is_to_aggregate;
   bool m_skip_last_line;
   std::string m_error_description;
-  char* m_stream;
-  char* m_end_stream;
+  std::stringstream m_stream;
+  std::stringstream m_end_stream;
   std::vector<char*> m_row_tokens{128};
   s3select* m_s3_select;
-  csvParser csv_parser;
+  int m_column_count = 1;
+  io::CSVReader<io::trim_chars<>, io::double_quote_escape<',', '\\'>, io::throw_on_overflow>* csv_parser;
```

Comment:
when an exception is raised? 
how it is handled? 
Compared to the previous parser (boost)  is there a change?